### PR TITLE
Fix unsafe pointer access pkg/githubclient/client_repository.go

### PR DESCRIPTION
**Pull request by TheoBrigitte**
Fix a bug where `devctl repo setup` fails with a panic when trying to get organization details for non org owned repositories.

```
$ devctl repo setup TheoBrigitte/test
INFO[0000] get repository details for "TheoBrigitte/test"
INFO[0000] configure repository settings
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x128e5a0]

goroutine 1 [running]:
github.com/giantswarm/devctl/pkg/githubclient.(*Client).SetRepositoryPermissions(0xc000433ee0?, {0x17928d8?, 0xc00004a150?}, 0xc00020d880?, 0xc0000ec700?)
        /home/theo/src/github.com/giantswarm/devctl/pkg/githubclient/client_repository.go:121 +0x40
github.com/giantswarm/devctl/cmd/repo/setup.(*runner).run(0xc000470960, {0x17928d8, 0xc00004a150}, 0x0?, {0xc0003b6ef0, 0x0?, 0x0?})
        /home/theo/src/github.com/giantswarm/devctl/cmd/repo/setup/runner.go:90 +0x34a
github.com/giantswarm/devctl/cmd/repo/setup.(*runner).Run(0x0?, 0x0?, {0xc0003b6ef0?, 0x0?, 0x0?})
        /home/theo/src/github.com/giantswarm/devctl/cmd/repo/setup/runner.go:32 +0x3e
github.com/spf13/cobra.(*Command).execute(0xc00047f500, {0xc0003b6ec0, 0x1, 0x1})
        /home/theo/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:916 +0x862
github.com/spf13/cobra.(*Command).ExecuteC(0xc000486600)
        /home/theo/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
        /home/theo/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:968
main.mainE({0x405cbc?, 0xc00018df70?})
        /home/theo/src/github.com/giantswarm/devctl/main.go:48 +0x98
main.main()
        /home/theo/src/github.com/giantswarm/devctl/main.go:16 +0x2b
exit status 2
```

This PR mitigate the above issue by providing safe pointer access to repository details:

 * safe pointer: *repository.Name > repository.GetName()
 * safe pointer: *repository.Owner.Login > repository.GetOwner().GetLogin()
 * safe pointer: *repository.Organization.Login > repository..GetOrganization().GetLogin()
 * safe pointer: *repository.DefaultBranch > repository.GetDefaultBranch()

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

**Comment by TheoBrigitte** [comment]
Wait and propagate fixes after https://github.com/giantswarm/devctl/pull/530

**Review by hervenicol — COMMENTED** [review]
Just to be sure I understand it right:
After the change it's using copies, so there's no risk to update the original data in functions?

**Comment by TheoBrigitte** [comment]
> Just to be sure I understand it right: After the change it's using copies, so there's no risk to update the original data in functions?

No, we were already using copies as we dereference the values via `*`.
The problem here is that we dereference/access values directly without checking if any of the actual pointers contains something. Therefore when trying to access a value over a `nil` pointer leads to an invalid address error as the pointer contains no address.

Examples:
* unsafe pointer access: https://go.dev/play/p/0mTeTVFl8BM
* safe pointer access: https://go.dev/play/p/y3L6WMFxN5N


Also everything here is a pointer `repository` is a pointer to `github.Repository` and `Name` is a pointer to `string`, so we need to check that every pointer is not nil before access its value. But the library already does this for us https://github.com/google/go-github/blob/89472069d5a89418c8619932e639541357e82ca2/github/github-accessors.go#L14417-L14423

TL;DR always use access methods instead of pointers.

Note: the program now does not panic but return an error due to the empty string passed to the api, but I am not fixing this here.

```
$ devctl repo setup TheoBrigitte/test
INFO[0000] get repository details for "TheoBrigitte/test"
INFO[0000] configure repository settings
INFO[0001] grant permission on repository
Error: PUT https://api.github.com/orgs//teams/Employees/repos/TheoBrigitte/test: 404 Not Found []
        /home/theo/src/github.com/giantswarm/devctl/pkg/githubclient/client_repository.go:132
        /home/theo/src/github.com/giantswarm/devctl/cmd/repo/setup/runner.go:91
        /home/theo/src/github.com/giantswarm/devctl/cmd/repo/setup/runner.go:34
        /home/theo/src/github.com/giantswarm/devctl/main.go:50
exit status 2
```

**Review by marieroque — APPROVED** [review]
